lib/ansible/plugins/lookup/secret_server.py

```diff
@@ -0,0 +1,226 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    author:
+        - Michal Taratuta (@MichalTaratuta for Softcat.com)
+    lookup: secret_server
+    short_description: Get secret from a Secret Server.
+    description:
+        - Retrieves data from a Secret Server.
+    options:
+        terms:
+            description:
+                - The list of keys to lookup on the Secret Server.
+            type: list
+            elements: string
+            required: True
+        username:
+            description:
+                - Username for the Secret Server user.
+            default: None
+            type: string
+            env:
+                - name: SS_USER_ANSIBLE
+            required: true (either ENV or kwargs)
+        password:
+            description:
+                - Password for the Secret Server user.
+            default: None
+            type: string
+            env:
+                - name: SS_PASSWORD_ANSIBLE
+            required: true (either ENV or kwargs)
+        domain:
+            description:
+                - Domain for the organisation in the Secret Server.
+            default: None
+            type: string
+            env:
+                - name: SS_DOMAIN_ANSIBLE
+            required: false (required only with SAAS)
+        address:
+            description:
+                - FQDN / IP for the Secret Server.
+            default: None
+            type: string
+            env:
+                - name: SS_ADDRESS_ANSIBLE
+            required: true (either ENV or kwargs)
+        key:
+            description:
+                - Name of the key for which you want to obtain the value.
+            default: password
+            type: string
+            required: false
+'''
+
+EXAMPLES = '''
+    - name: "Get passowrd for customer_1 with ENV variables"
+      debug:
+        msg: "{{ lookup('secret_server', 'customer_1') }}"
+
+    - name: "Get Key=Username for customer_1 with ENV variables"
+      debug:
+        msg: "{{ lookup('secret_server', 'customer_1', key='username') }}"
+
+    - name: "Get passowrd for customer_1"
+      debug:
+        msg: >
+            "{{ lookup('secret_server', 'customer_1',
+            address='server1.example.com', domain='example.com',
+            username='awesome_user', password='awesome_password') }}"
+'''
+
+RETURN = '''
+    raw:
+        description:
+            - List of values associated with input keys.
+        type: list
+        elements: strings
+'''
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+import os
+
+try:
+    import requests
+    import urllib3
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+# Check if ENV variables are present
```

Comment:
Env vars should be handled via `self.get_option` instead of independent env var fetching.  This allows proper config file use from the options defined in your `DOCUMENTATION`

Reply:
@sivel I'm new to python and I couldn't make this working, am I right that this would just pull the `default` configuration from the `DOCUMENTATION`?